content/docs/08-troubleshooting/01-migration-guide/01-migration-guide-4-0.mdx

```diff
@@ -226,6 +321,11 @@ The `warnings` property of the `StreamObjectResult` type is now a Promise.
 
 ## AI SDK RSC Changes
 
+<Note type="warning">
```

Comment:
In the interest of reduced verbosity I made the design decision to put this header in sections where all of the items in that section don't have codemods. The tradeoff is that someone jumping to look at just one specific item in a large section may not scroll up and see this in the section header.

I'm open to putting a note in every item rather than rolling up to sections if preferred.

Reply:
I think I'm leaning towards being explicit where we can to avoid the potential issues from what you mentioned.

Reply:
no strong opinion either way.

Reply:
I will leave it by section for now.